File: leetcode/面试题02.07.链表相交/0207.cpp
# Definition for singly - linked list.
# class ListNode :
    #     def __init__(self, x) :
    #         self.val = x
    #         self.next = None

    class Solution :
    def getIntersectionNode(self, headA: ListNode, headB : ListNode)->ListNode :
    curA = headA
    curB = headB
    lenA = 0
    lenB = 0
    while curA :
        lenA += 1
        curA = curA.next
        while curB :
            lenB += 1
            curB = curB.next
            curA = headA
            curB = headB
            if lenB > lenA:
tmp = lenA
lenA = lenB
lenB = tmp
tmpnode = curA
curA = curB
curB = tmpnode
gap = lenA - lenB
for i in range(gap) :
    curA = curA.next
    gap -= 1
    while curA :
        if curA == curB :
            return curA
            curA = curA.next
            curB = curB.next
            return None

File: leetcode/面试题02.07.链表相交/0207.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode curA = headA;
        ListNode curB = headB;
        int lenA = 0, lenB = 0;
        while(curA != null){
            lenA++;
            curA = curA.next;
        }
        while(curB != null){
            lenB++;
            curB = curB.next;
        }
        curA = headA;
        curB = headB;
        if(lenB > lenA){
            int tmp = lenA;
            lenA = lenB;
            lenB = tmp;
            ListNode Tmp = curA;
            curA = curB;
            curB = Tmp;
        }
        int gap = lenA - lenB;
        while(gap-- > 0){
            curA = curA.next;
        }
        while(curA != null){
            if(curA == curB){
                return curA;
            }
            curA = curA.next;
            curB = curB.next;
        }
        return null;
    }
}